internal/build/job/maker.go

```diff
@@ -18,13 +14,62 @@ import (
 	"k8s.io/utils/pointer"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/build"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/constants"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/utils"
 )
 
+const (
+	dockerfileVolumeName = "dockerfile"
+)
+
+type jobTemplateOptions struct {
+	containerImage string
+
+	registryTLS *kmmv1beta1.TLSOptions
+
+	owner metav1.Object
+}
+
+// JobTemplateOption configures MakeJobTemplate.
+type JobTemplateOption func(*jobTemplateOptions)
+
+// ContainerImage configures the image to be pushed. When left
+// empty no image is pushed.
+func ContainerImage(containerImage string) JobTemplateOption {
+	return func(opts *jobTemplateOptions) {
+		opts.containerImage = containerImage
+	}
+}
+
+// RegistryTLS configures the TLSOptions to be used when
+// interacting with the registry.
+func RegistryTLS(registryTLS *kmmv1beta1.TLSOptions) JobTemplateOption {
```

Comment:
is this functions used somewhere beside job package? If not, then it should be private